app/policies/collection_policy.rb

```diff
@@ -13,8 +13,8 @@ def new?
     everyone
   end
 
-  def show?
-    admin? || collection.users.include?(@user) || collection.visibility_level_public?
+  %i[show? toggle_favorite?].each do |action|
```

Comment:
I also prefer duplication to meta programming for simplicity, but that's also rather a personal aspect.

Reply:
Normally I agree and tend to avoid meta programming, but in case of these policies I like that you can easily see which actions belong together. Especially if you have a lot of actions (like in line 20-21 here)